apps/functions-subscription/src/adapters/azure/functions/middleware.ts

```diff
@@ -33,12 +35,31 @@ export const parsePathParameter =
       E.mapLeft(({ message }) => new H.HttpBadRequestError(message)),
     );
 
+/**
+ * Parses a specific header parameter of an HTTP request using the provided schema.
+ */
+export const parseHeaderParameter =
+  <T>(schema: Decoder<unknown, T>, paramName: string) =>
+  (req: H.HttpRequest) =>
+    pipe(
+      req.headers[paramName],
+      E.fromNullable(new H.HttpBadRequestError(`Missing ${paramName} header`)),
```

Comment:
**note**: now, if the header is not in the request, the message `Missing <param> header` is returned to the caller within a `400 Bad Request` response.